File: packages/go.dev/libraries/utils/lodash/number/number.go
package number

import "math/rand"

func Clamp(number float64, lower float64, upper float64) float64 {
	if number < lower {
		return lower
	}
	if number > upper {
		return upper
	}
	return number
}

func InRange(params ...float64) bool {
	if len(params) == 2 {
		var number float64 = params[0]
		var stop float64 = params[1]
		return 0 < number && number < stop
	}

	if len(params) >= 3 {
		var number float64 = params[0]
		var start float64 = params[1]
		var stop float64 = params[2]
		return start < number && number < stop
	}

	return false
}

func Random(lower int, upper int) int {
	if lower > upper {
		var temp int = lower
		lower = upper
		upper = temp
	}
	return rand.Intn(upper-lower) + lower
}

File: packages/python.org/lodash/libs/number.py
"""
number
"""

import random as rand


def clamp(number, lower, upper):
    """
    clamp
    """
    if lower > upper:
        return
    if number < lower:
        return lower
    elif number > upper:
        return upper
    else:
        return number


def in_range(number, x_value, y_value=None):
    """
    in_range
    """
    start = 0
    stop = x_value
    if y_value is not None:
        start = x_value
        stop = y_value
    return start < number and number < stop


def random(x_value, y_value=None, step=1):
    """
    random
    """
    start = 0
    end = x_value
    if y_value is not None:
        start = x_value
        end = y_value
    return rand.randrange(start, end, step)
